src/EditorFeatures/CSharpTest/CodeRefactorings/AddMissingImports/AddMissingImportsRefactoringProviderTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.UnitTests;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.PasteTracking;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.Composition;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.AddMissingImports
+{
+    [UseExportProvider]
+    [Trait(Traits.Feature, Traits.Features.AddMissingImports)]
+    public class AddMissingImportsRefactoringProviderTests : AbstractCSharpCodeActionTest
```

Comment:
The AddMissingImportsFeatureService tests ensure that missing imports are being added as expected. The RefactoringProvider tests ensure that we are offering the refactoring as expected. I could imagine an end to end test that validates both, but it may be more complicated then testing each separately.

Reply:
in general, we do not test the intermediary components of a feature.  That's because we're then effectively testing internal implementation details of the feature, and we're make a brittle and harder to maintain connection between the tests and the impl.  In general, we want to write the tests 'above' the entire feature, and then validate the user facing experience.  That gives us maximal ability to reimplement things however we want, without having to do much with the tests.

You'll note that effectively no other refactorings/fixes tests have this sort of split, even though many of them are built up out of many subcomponents.